src/books/repositories/typeorm/book-repository-typeorm.adapter.ts

```diff
@@ -1,40 +1,132 @@
+    const allBooks: Book[] = await this.booksRepository.find();
+    await this.bookModel.deleteMany({});
+    for (let index = 0; index < allBooks.length; index++) {
+      const element: Book = allBooks[index];
+      const createdBook: BookMongoDocument = new this.bookModel(element);
+      await createdBook.save();
+    }
+  }
+
   async findOneById(id: string): Promise<Book> {
-    return this.booksRepository.findOne(id);
+    // return this.booksRepository.findOne(id);
+    const document: BookMongoDocument = await this.bookModel.findOne({ id: id }).exec();
+    if (document === null) {
+      return undefined;
+    } else {
+      return new Book(document.toObject());
+    }
   }
 
   async findAll(): Promise<Book[]> {
-    return this.booksRepository.find();
+    // return this.booksRepository.find();
+
+    // Je crois que je me suis perdu, help pour reecrire avec async/await
+    // return new Promise((resolve, reject) => {
+    //   this.bookModel
+    //     .find({})
+    //     .exec()
+    //     .then((value: BookMongoDocument[]) => {
+    //       const tmp: Book[] = [];
+    //       for (let index = 0; index < value.length; index++) {
+    //         tmp[index] = new Book(value[index].toObject());
+    //       }
+    //       resolve(tmp);
+    //     })
+    //     .catch((error) => {
+    //       reject(error);
+    //     });
+    // });
+
+    const documents: BookMongoDocument[] = await this.bookModel.find({}).exec();
+    const output: Book[] = [];
+    for (let index = 0; index < documents.length; index++) {
+      output[index] = new Book(documents[index].toObject());
+    }
+    return output;
+  }
+
+  private async handleCreateMongo(id: string): Promise<void> {
+    const toWrite: Book = await this.booksRepository.findOne(id);
+    const createdBook: BookMongoDocument = new this.bookModel(toWrite);
+    //? await or not for last ? or return ? or return await ?
+    await createdBook.save();
   }
 
   async create(id: string, payload: CreateBookDto): Promise<Book> {
-    return this.booksRepository.save({
+    const output: Promise<Book> = this.booksRepository.save({
       id,
       ...payload,
     });
+    //? Possible de reecrire avec async ???
+    output.then(async () => {
+      await this.handleCreateMongo(id);
+    });
+    return output;
+  }
+
+  private async handleDeleteMongo(id: string): Promise<void> {
+    //? await or not ?
+    await this.bookModel.deleteOne({ id: id });
   }
 
   async delete(id: string): Promise<void> {
-    await this.booksRepository.delete(id);
+    // The real type is Promise<DeleteResult>
+    //  not used to avoid useless import
+    //  and beacause the type is not relevant in this case
+    const output: Promise<any> = this.booksRepository.delete(id);
+    output.then(async () => {
+      await this.handleDeleteMongo(id);
+    });
+    //? ou juste 'return' ?
+    return output;
```

Comment:
Je suis d'accord avec tes commentaires, dans ce cas on peut juste retourner void nous meme une fois l'action terminée comme tu voulais !
Et voila une suggestion pour async/await les promises et retourner void :)
```suggestion
    const output = await this.booksRepository.delete(id);
    await this.handleDeleteMongo(id);
    return;
```